pkg/controller/worker/machines.go

```diff
@@ -177,6 +177,12 @@ func (w *workerDelegate) generateMachineConfig(_ context.Context) error {
 		}
 
 		gceInstanceLabels := getGceInstanceLabels(w.worker.Name, pool)
+
+		if pool.MachineImage.Name != "" && pool.MachineImage.Version != "" {
```

Comment:
Why not move this code into function `getGceInstanceLabels` itself  ? anyways `MachineImage/Version` fields are part of `pool` obj which is passed to `getGceInstanceLabels`